corehq/apps/userreports/views.py

```diff
@@ -213,6 +213,17 @@ def section_url(self):
     def page_url(self):
         return reverse(self.urlname, args=(self.domain,))
 
+    def dispatch(self, *args, **kwargs):
+        allow_access_to_ucrs = (
+            self.request.couch_user.has_permission(
+                self.domain,
+                get_permission_name(Permissions.edit_ucrs)
+            )
+        )
+        if allow_access_to_ucrs or not settings.IS_SAAS_ENVIRONMENT:
```

Comment:
will this always show the view if it's not a SAAS environment? Is that what we want? I think in a non-saas environment we still want to handle permissions by user... but maybe I missed a more in depth discussion here.

Reply:
Yeah. You are correct Biyeun. I missed this check while updating the code. Will fix it now.

Reply:
Done in https://github.com/dimagi/commcare-hq/pull/30439/commits/03040add14bf8acb8161c50157548ef92abf93ae